src/services/rfqm_service.ts

```diff
@@ -513,6 +528,184 @@ export class RfqmService {
+        while (!isTxFinalized) {
+            await delay(TRANSACTION_WATCHER_SLEEP_TIME_MS);
+
+            const statusCheckResult = await this._checkSubmissionMapReceiptsAndUpdateDbAsync(
+                submissionsMap,
+                expectedTakerTokenFillAmount,
+            );
+            isTxMined = statusCheckResult.isTxMined;
+            isTxFinalized = statusCheckResult.isTxFinalized;
+            submissionsMap = statusCheckResult.submissionsMap;
+
+            if (!isTxMined) {
+                const newGasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
+
+                if (gasPrice.lt(newGasPrice)) {
+                    gasPrice = newGasPrice;
+                    const submission = await this._submitTransactionAsync(
+                        orderHash,
+                        workerAddress,
+                        callData,
+                        gasPrice,
+                        nonce,
+                        gasEstimate,
+                        expectedTakerTokenFillAmount,
+                    );
+                    submissionsMap[submission.transactionHash!] = submission;
+                }
+            }
+        }
+    }
+
+    /**
+     * Check for receipts from the tx's in a SubmissionsMap object
+     * Update database with status of all tx's
+     */
+    private async _checkSubmissionMapReceiptsAndUpdateDbAsync(
+        submissionsMap: SubmissionsMap,
+        expectedTakerTokenFillAmount: BigNumber,
+    ): Promise<SubmissionsMapStatus> {
+        let isTxMined: boolean = false;
+        let isTxFinalized: boolean = false;
+
+        // check if any tx has been mined
+        const receipts = await Promise.all(
+            Object.keys(submissionsMap).map(async (transactionHash) => {
+                return {
+                    transactionHash,
+                    response: await this._blockchainUtils.getTransactionReceiptIfExistsAsync(transactionHash),
+                };
+            }),
+        );
+
+        for (const receipt of receipts) {
+            if (receipt.response !== undefined) {
+                isTxMined = true;
+                const currentBlock = await this._blockchainUtils.getCurrentBlockAsync();
+                if (receipt.response.blockNumber - currentBlock >= BLOCK_FINALITY_THRESHOLD) {
+                    isTxFinalized = true;
+                }
+                // update all entities
+                // since the same nonce is being re-used, we expect only 1 defined receipt
+                for (const r of receipts) {
+                    if (r.response !== undefined) {
+                        if (r.response.status === 1) {
+                            const actualTakerTokenFilledAmount = this._blockchainUtils.getRfqOrderTakerTokenFilledAmountFromLogs(
+                                r.response.logs,
+                            );
+                            submissionsMap[r.transactionHash].status = actualTakerTokenFilledAmount.lt(
```

Comment:
small suggestion, in JS all objects are references unless explicitly copied, so you can avoid the redundant `submissionsMap[r.transactionHash]` with an interin variable assignment.

```
const successfulTransaction = submissionsMap[r.transactionHash];
successfulTransaction.status = ...
successfulTransaction.actualTakerTokenFillAmount = ....
```